src/Compilers/Core/Portable/Compilation/IImportScope.cs

```diff
@@ -0,0 +1,144 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+
+namespace Microsoft.CodeAnalysis;
+
+/// <summary>
+/// Represents the set of symbols that are imported to a particular position in a source file. Each import has a
+/// reference to the location the import directive was declared at.  For the <see cref="IAliasSymbol"/> import, the
+/// location can be found using either <see cref="ISymbol.Locations"/> or <see
+/// cref="ISymbol.DeclaringSyntaxReferences"/> on the <see cref="IAliasSymbol"/> itself.  For <see cref="Imports"/>
+/// or <see cref="XmlNamespaces"/> the location is found through <see
+/// cref="ImportedNamespaceOrType.DeclaringSyntaxReference"/> or <see
+/// cref="ImportedXmlNamespace.DeclaringSyntaxReference"/> respectively.
+/// </summary>
+/// <remarks>
+/// <list type="bullet">
+/// <item>Scopes returned will always have at least one non-empty property value in them.</item>
+/// <item>Symbols may be imported, but may not necessarily be available at that location (for example, an alias
+/// symbol hidden by another symbol).</item>
+/// <item>
+/// In C# there will be an <see cref="IImportScope"/> for every containing namespace-declarations that include any
+/// import directives.  There will also be an <see cref="IImportScope"/> for the containing compilation-unit if it
+/// includes any import directives or if there are global import directives pulled in from other files.
+/// </item>
+/// <item>
+/// In Visual Basic there will only be at most two <see cref="IImportScope"/>s returned for any position.  There
```

Comment:
>here will only be at most two <see cref="IImportScope"/>s returned for any position. [](http://example.com/codeflow?start=21&length=84)

Is this accurate for scripting scenarios, if we support them for VB? Imports from previous submissions, etc.

Reply:
I am specifically referring to the "at most two" part. Perhaps we should soften this.

Reply:
Sure :)

Reply:
Relaxed this.